api/src/modules/User/resolvers.ts

```diff
@@ -12,24 +12,21 @@ export const me = queryField('me', {
   },
 });
 
-export const signIn = mutationField('signIn', {
-  type: 'AuthPayload',
-  args: {
-    user: nonNull('UserNamePass'),
-  },
-  resolve: async (_parent, { user: userNamePass }, ctx) => {
-    const { id } = await ctx.prisma.user.getUserByUsernamePassword(userNamePass);
-    return ctx.prisma.session.createSignInSession(id);
-  },
+export const getChallenge = mutationField('getChallenge', {
+  type: 'Challenge',
+  resolve: async (_, __, ctx) => {
+    return ctx.prisma.session.getChallenge();
+  }
```

Comment:
Instead of returning a single id from this endpoint and ask users to generate complete message on their own, I would return a complete message to be singed as a basic string